resilience4j-kotlin/src/main/kotlin/io/github/resilience4j/kotlin/bulkhead/Bulkhead.kt

```diff
@@ -44,4 +52,20 @@ suspend fun <T> Bulkhead.executeSuspendFunction(block: suspend () -> T): T {
  */
 fun <T> Bulkhead.decorateSuspendFunction(block: suspend () -> T): suspend () -> T = {
     executeSuspendFunction(block)
+}
+
+/**
+ * Try to immediately acquire permission from the bulkhead when it is not expected to block.
+ *
+ * If a wait duration is configured on the bulkhead, then attempt to acquire permission within
+ * the confines of a dispatcher specialized for blocking calls.
+ *
+ */
+internal suspend fun Bulkhead.acquirePermissionSuspend(){
+    // Fast path. Avoid dispatch context switch.
+    if(bulkheadConfig.maxWaitDuration.isZero){
+        acquirePermission()
+    }else{
+        withContext(Dispatchers.IO){ acquirePermission() }
+    }
```

Comment:
This is being used to mitigate the risks associated with `acquirePermission` being a blocking API